substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -2146,6 +2180,19 @@ pub mod pallet {
 			);
 			Ok(())
 		}
+
+		#[pallet::call_index(30)]
+		#[pallet::weight(T::WeightInfo::migrate_currency())]
+		pub fn migrate_currency(
+			origin: OriginFor<T>,
+			stash: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+			Self::do_migrate_currency(&stash)?;
+
+			// Refund the transaction fee if successful.
+			Ok(Pays::No.into())
```

Comment:
I've checked this in your code already, but important to ensure `do_migrate_currency` errors on duplicates, to not make this abuse-able.

Reply:
added in test